### PR TITLE
CASMTRIAGE-7420 Fix kdump hook boot failure

### PR DESCRIPTION
**Pull request by rustydb**
### Summary and Scope

- Fixes: CASMTRIAGE-7420
- Relates to: #66 

#### Issue Type

- Bugfix Pull Request

Fixes issues when booting without an overlayFS.

Removes fallback assumptions around FSLabels that were causing code to run in undesirable cases.

### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (if yes, please include results or a description of the test)
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)
 
### Risks and Mitigations
 

Successfully tested deploying and rebooting:
- NCN-Master
- NCN-Worker
- NCN-Storage

Successfully triggered a crash on:
- NCN-Kubernetes
- NCN-Storage

> NOTE: RemoteISO panics are not supported
> NOTE2: USB ISO panics are broken, USB devices are not detected in the kdump initrd.

### CODE REVIEW

**Review by jpdavis-prof — APPROVED** [review]

File: 90metalmdsquash/metal-md-lib.sh [hunk 1]
```diff
@@ -221,17 +181,21 @@ add_overlayfs() {
...
     # try shasta-1.3 formatting or die.
     mount -n -t ext4 /dev/md/ROOT "$mpoint" \
-      || metal_die "Failed to mount ${oval_drive_authority} as xfs or ext4"
+      || metal_die "Failed to mount /dev/md/ROOT as xfs or ext4"
   fi
 
   # Create OverlayFS directories for dmsquash-live
   # See source-code for details: https://github.com/dracutdevs/dracut/blob/09a1e5afd2eaa7f8e9f3beaf8a48283357e7fea0/modules.d/90dmsquash-live/dmsquash-live-root.sh#L168-L169
   # Requires two directories; ovlwork, and overlay-$FSLABEL-$UUID (where FSLABEL and UUID are of the partition containing the squashFS image).
-  [ -z "${METAL_OVERLAYfs_id}" ] && METAL_OVERLAYfs_id="$(_overlayFS_path_spec)"
```
Comment (rustydb) [inline 90metalmdsquash/metal-md-lib.sh:230]: Need to break this out into a separate commit.

**Comment by rustydb** [comment]
Using surtur for a remoteISO test.
Will use redbull for testing k8s and a usb-pit.
